tests/backward-compat-upgrade-direct/pom.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="
+  http://maven.apache.org/POM/4.0.0
+  http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.bookkeeper.tests</groupId>
+    <artifactId>integration-tests-base-groovy</artifactId>
```

Comment:
any reason why this module's parent is "integration-tests-base-groovy"? can't we just include `integration-tests-base-groovy` as a dependency?

it is also an interesting layout to me. I think we typically organize the parent module in in hierarchical. it seems that you are using a sibling module as the parent. any reason? it might be good to follow how other modules are organized and make it consistent across the project.

it probably can be organized in this way:

- `tests/integration-tests`
- `tests/integration-tests/base`
- `tests/integration-tests/base-groovy`
- `tests/integration-tests/utils`
- `tests/integration-tests/topologies`
- `tests/integration-tests/backward-compat`

Reply:
I used a parent instead of dependency, because i want plugin settings to be inherited too, such as the groovy stuff and surefire stuff. I can move them around to change the structure if you like. Could we do it after the whole series is in though?

Reply:
sure. it's better to organize them in a better hierarchy, so people is easy to understand and maintain. please create an issue for it.

Reply:
#1030 created. We need to start triaging issues so this stuff gets picked up.